Jenkinsfile.treecompose

```diff
@@ -92,7 +94,9 @@ node(NODE) {
             version = utils.get_rev_version("${repo}", commit)
             currentBuild.description = "🆕 ${version} (commit ${commit})";
             if (previous_commit.length() > 0) {
-                sh "rpm-ostree --repo=${repo} db diff ${previous_commit} ${commit}"
+                sh "mkdir -p ${images}"
+                utils.rsync_dir_in(ARTIFACT_SERVER, KEY_FILE, images)
+                sh "rpm-ostree --repo=${repo} db diff ${previous_commit} ${commit} > /${images}/pkg_diff.txt"
```

Comment:
I think we should archive the `pkg_diff.txt` each time it is created, as it will be overwritten whenever there is a new commit to diff.

Perhaps you can write out the diff to `${WORKSPACE}/pkg_diff.txt` and then copy it to `${images}`.  Having the file in the workspace would allow us to use the `archiveAritifacts()` function like we do in the cloud pipeline:

https://github.com/openshift/os/blob/master/Jenkinsfile.cloud#L94-L96

Reply:
Updated. If we archive pkg_diff.txt, doesn't it still get overwritten?

Reply:
My understanding of `archiveArtirfacts` is that it keeps the files around on the Jenkins host until they are pruned.  For example, in the `cloud` pipeline, we archive the `rhcos-qcow2-install.txt`:

https://continuous-infra-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/rhcos/job/coreos-rhcos-cloud/1561/artifact/rhcos-qcow2-install.txt

https://continuous-infra-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/rhcos/job/coreos-rhcos-cloud/1558/artifact/rhcos-qcow2-install.txt

So we are able to keep files per build, but it is overwritten on the artifact server in the `sync out` stage.